src/main/java/io/debezium/connector/yugabytedb/YugabyteDBValueConverter.java

```diff
@@ -401,10 +400,15 @@ public ValueConverter converter(Column column, Field fieldDefn) {
             case PgOid.NUMERIC:
                 return (data) -> convertDecimal(column, fieldDefn, data, decimalMode);
             case PgOid.BYTEA:
-                return data -> convertString(column, fieldDefn, data);
-            // Commented out because bytea are converted to strings as of now and being sent
-            // across Debezium
-            // return data -> convertBinary(column, fieldDefn, data, binaryMode);
+                return data -> {
+                    // Convert to byte array if data is a hexadecimal string.
+                    if (data instanceof String && ((String) data).startsWith("\\x")) {
+                        byte[] dataBytes = HexConverter.convertFromHex(((String) data).substring(2));
+                        return convertBinary(column, fieldDefn, dataBytes, binaryMode);
+                    }
```

Comment:
I added this logic here instead of overriding the underlying functions because there were multiple functions that were dependent on the type of `data`.